utils/s3/client.cc

```diff
@@ -223,6 +234,92 @@ future<client::stats> client::get_object_stats(sstring object_name) {
     co_return st;
 }
 
+static rapidxml::xml_node<>* first_node_of(rapidxml::xml_node<>* root,
+                                           std::initializer_list<std::string_view> names) {
+    assert(root);
+    auto* node = root;
+    for (auto name : names) {
+        node = node->first_node(name.data(), name.size());
+        if (!node) {
+            throw std::runtime_error(fmt::format("'{}' is not found", name));
+        }
+    }
+    return node;
+}
```

Comment:
note to myself, we can reuse this helper in other places in this source file where we want to reference an element with XPath like 'foo/bar'. despite that the API document might claim that there will be one or more bar under `foo`, we should handle the cases where `foo` or `foo/bar` does not exist, and error out if it really happens.